easybuild/easyblocks/c/code_saturne.py

```diff
@@ -0,0 +1,167 @@
+            'slurm': [False, "Build for the slurm workload manager.", CUSTOM],
+        })
+
+        return extra_vars
+
+    def configure_step(self):
+        """Configure step for Code_Saturne."""
+
+        self.log.info("Configuration step is running...")
+
+        # only use the opt flags
+        env.setvar("CFLAGS", os.environ['OPTFLAGS'])
+        env.setvar("CXXFLAGS", os.environ['OPTFLAGS'])
+        env.setvar("FCFLAGS", os.environ['OPTFLAGS'])
+
+        self.log.info("Running ./sbin/bootstrap ...")
+        cmd = './sbin/bootstrap'
+        run_cmd(cmd, log_all=True, simple=True, log_output=True)
+
+        cmd = ' '.join([
+            self.cfg['preconfigopts'],
+            './configure',
+            '--prefix=' + self.installdir,
+            '--without-modules',
+            self.cfg['configopts'],
+            ])
+
+        if self.cfg['debug']:
+            cmd = ' '.join([cmd, '--enable-debug'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def build_step(self):
+        """ Build step for Code_Saturne."""
+
+        paracmd = ''
+        if self.cfg['parallel']:
+            paracmd = "-j %s" % self.cfg['parallel']
+
+        cmd = "%s make %s %s" % (self.cfg['prebuildopts'],
+                                 paracmd, self.cfg['buildopts'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def install_step(self):
+        """ Install step for Code_Saturne."""
+
+        cmd = "%s make install %s" % (self.cfg['preinstallopts'],
+                                      self.cfg['installopts'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def post_install_step(self):
+        """Custom post install step for Code_Saturne."""
+        super(EB_Code_underscore_Saturne, self).post_install_step()
+
+        # if we are using a SLURM system, we need to change the defaults;
+        # so, modify the template and rename it to "etc/code_saturne.cfg"
+        if self.cfg['slurm']:
+
+            self.log.info("Running the post-install SLURM step ...")
+
+            target_path = os.path.join(self.installdir, 'etc/code_saturne.cfg')
+            from_path = target_path + '.template'
+
+            apply_regex_substitutions(
+                from_path,
+                [(r"# batch =", r"batch = SLURM")]
+                )
+            apply_regex_substitutions(
+                from_path,
+                [(r"# mpiexec = mpiexec", r"mpiexec = srun")]
+                )
+            apply_regex_substitutions(
+                from_path,
+                [(r"# mpiexec_n = ' -n '", r"mpiexec_n = ' -n '")]
+                )
+            apply_regex_substitutions(
+                from_path,
+                [(r"# mpiexec_n_per_node =",
+                  r"mpiexec_n_per_node = ' --ntasks-per-node '")]
+                )
```

Comment:
The multi-line formatting is a bit over the top here imho...

Also, why not make this a single call to `apply_regex_substitutions`?

```python
regex_subs = [
    (r"# batch =", r"batch = SLURM")],
    (r"# mpiexec = mpiexec", r"mpiexec = srun"),
    (r"# mpiexec_n = ' -n '", r"mpiexec_n = ' -n '"),
    (r"# mpiexec_n_per_node =", r"mpiexec_n_per_node = ' --ntasks-per-node '"),
]
apply_regex_substitutions(from_path, regex_subs)